subsystems/win32/csrsrv/api.c

```diff
@@ -189,7 +189,7 @@ CsrApiHandleConnectionRequest(IN PCSR_API_MESSAGE ApiMessage)
     ConnectInfo->ServerProcessId = NtCurrentTeb()->ClientId.UniqueProcess;
 
     /* Accept the Connection */
-    ASSERT(!AllowConnection || (AllowConnection && CsrProcess));
+    ASSERT(!AllowConnection || CsrProcess);
```

Comment:
I think this makes the assertion condition less readable.

Reply:
It is clear that if first condition is false, `AllowConnection` is true. No need to check bool value again...